File: ToDoApp/ToDoApp.Web/Modules/ServerTypes/TaskDB/TaskForm.ts
import { StringEditor, TextAreaEditor, EnumEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { Priority } from "../Modules/TaskDB.Task.Priority";
import { Status } from "../Modules/TaskDB.Task.Status";

export interface TaskForm {
    Title: StringEditor;
    Description: TextAreaEditor;
    Status: EnumEditor;
    Priority: EnumEditor;
    DueDate: DateEditor;
}

export class TaskForm extends PrefixedContext {
    static readonly formKey = 'TaskDB.Task';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TaskForm.init)  {
            TaskForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = EnumEditor;
            var w3 = DateEditor;

            initFormType(TaskForm, [
                'Title', w0,
                'Description', w1,
                'Status', w2,
                'Priority', w2,
                'DueDate', w3
            ]);
        }
    }
}

[Status, Priority]; // referenced types

File: ToDoApp/ToDoApp.Web/Modules/TaskDB/Task/TaskForm.cs
using Serenity.ComponentModel;
using ToDoApp.Modules.TaskDB.Task;

namespace ToDoApp.TaskDB.Forms;

[FormScript("TaskDB.Task")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskForm
{
    public string Title { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    public Status Status { get; set; }
    public Priority Priority { get; set; }
    public DateTime DueDate { get; set; }
}